### PR TITLE
doc: Adds deprecation banner to `mongodbatlas_cluster` resource and data sources

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Adds deprecation banner to `mongodbatlas_cluster` resource and data sources

Link to any related issue(s): CLOUDP-261481

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [x] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

**Comment by lantoli** [comment]
@EspenAlbert it's assumed that the guide will in file cluster-to-advanced-cluster-migration-guide.md

File: docs/data-sources/cluster.md [hunk 1]
```diff
@@ -1,14 +1,20 @@
+---
+subcategory: "Deprecated"    
+---
+
+**WARNING:** This data source is deprecated, use `mongodbatlas_advanced_cluster`. In order to learn more about how to migrate, please read the [Migration Guide](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/cluster-to-advanced-cluster-migration-guide).
```
Comment (carriecwk) [inline docs/data-sources/cluster.md]: Succint phrasing: "To learn more about how to migrate, please read the [Migration Guide](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/cluster-to-advanced-cluster-migration-guide)."

File: docs/data-sources/clusters.md [hunk 2]
```diff
@@ -1,14 +1,20 @@
+---
+subcategory: "Deprecated"    
+---
+
+**WARNING:** This data source is deprecated, use `mongodbatlas_advanced_clusters`. In order to learn more about how to migrate, please read the [Migration Guide](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/cluster-to-advanced-cluster-migration-guide).
```
Comment (carriecwk) [inline docs/data-sources/clusters.md]: Succint phrasing: "To learn more about how to migrate, please read the [Migration Guide](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/cluster-to-advanced-cluster-migration-guide)."

File: docs/guides/1.18.0-upgrade-guide.md [hunk 3]
```diff
@@ -14,6 +14,8 @@ The Terraform MongoDB Atlas Provider version 1.18.0 has a number of new and exci
 
 **Deprecations:**
 
+- `mongodbatlas_cluster` resource and corresponding data sources have been deprecated in favor of `mongodbatlas_advanced_cluster`. In order to learn more about how to migrate, please read the [Migration Guide](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/cluster-to-advanced-cluster-migration-guide).
```
Comment (carriecwk) [inline docs/guides/1.18.0-upgrade-guide.md]: Succint phrasing: "To learn more about how to migrate, please read the [Migration Guide](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/cluster-to-advanced-cluster-migration-guide)."

File: docs/resources/cluster.md [hunk 4]
```diff
@@ -1,7 +1,16 @@
+---
+subcategory: "Deprecated"    
+---
+
+**WARNING:** This resource is deprecated, use `mongodbatlas_advanced_cluster`. In order to learn more about how to migrate, please read the [Migration Guide](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/cluster-to-advanced-cluster-migration-guide).
```
Comment (carriecwk) [inline docs/resources/cluster.md]: Succint phrasing: "To learn more about how to migrate, please read the [Migration Guide](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/cluster-to-advanced-cluster-migration-guide)."

File: docs/resources/cluster.md [hunk 5]
```diff
@@ -1,7 +1,16 @@
...
+
 # Resource: mongodbatlas_cluster
 
 `mongodbatlas_cluster` provides a Cluster resource. The resource lets you create, edit and delete clusters. The resource requires your Project ID.
 
+~> **IMPORTANT:**
+<br> &#8226; New Users: If you are not already using `mongodbatlas_cluster` for your deployment we recommend starting with the [`mongodbatlas_advanced_cluster`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/advanced_cluster).  `mongodbatlas_advanced_cluster` has all the same functionality as `mongodbatlas_cluster` but also supports multi-cloud clusters.
```
Comment (carriecwk) [inline docs/resources/cluster.md]: Succint phrasing: `mongodbatlas_advanced_cluster` has the same functionality as `mongodbatlas_cluster` but also supports multi-cloud clusters.

File: examples/mongodbatlas_cluster/nvme-upgrade/README.md [hunk 6]
```diff
@@ -1,4 +1,7 @@
 # MongoDB Atlas Provider -- Cluster NVME (Non-Volatile Memory Express) Upgrade
+
+**WARNING:** `mongodbatlas_cluster` resource and corresponding data sources have been deprecated in favor of `mongodbatlas_advanced_cluster`. In order to learn more about how to migrate, please read the [Migration Guide](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/cluster-to-advanced-cluster-migration-guide).
```
Comment (carriecwk) [inline examples/mongodbatlas_cluster/nvme-upgrade/README.md]: Succint phrasing: "To learn more about how to migrate, please read the [Migration Guide](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/cluster-to-advanced-cluster-migration-guide)."

File: examples/mongodbatlas_cluster/tenant-upgrade/README.md [hunk 7]
```diff
@@ -1,4 +1,7 @@
 # MongoDB Atlas Provider -- Cluster Tenant Upgrade
+
+**WARNING:** `mongodbatlas_cluster` resource and corresponding data sources have been deprecated in favor of `mongodbatlas_advanced_cluster`. In order to learn more about how to migrate, please read the [Migration Guide](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/cluster-to-advanced-cluster-migration-guide).
```
Comment (carriecwk) [inline examples/mongodbatlas_cluster/tenant-upgrade/README.md]: Succint phrasing: "To learn more about how to migrate, please read the [Migration Guide](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/cluster-to-advanced-cluster-migration-guide)."

**Review by carriecwk — COMMENTED** [review]
Some comments!

File: docs/data-sources/cluster.md (hunk 1 above)
Comment (lantoli) [reply docs/data-sources/cluster.md]: thanks, changed

File: docs/data-sources/clusters.md (hunk 2 above)
Comment (lantoli) [reply docs/data-sources/clusters.md]: thanks, changed

File: docs/guides/1.18.0-upgrade-guide.md (hunk 3 above)
Comment (lantoli) [reply docs/guides/1.18.0-upgrade-guide.md]: thanks, changed

File: docs/resources/cluster.md (hunk 4 above)
Comment (lantoli) [reply docs/resources/cluster.md]: thanks, changed

File: examples/mongodbatlas_cluster/nvme-upgrade/README.md (hunk 6 above)
Comment (lantoli) [reply examples/mongodbatlas_cluster/nvme-upgrade/README.md]: thanks, changed

File: examples/mongodbatlas_cluster/tenant-upgrade/README.md (hunk 7 above)
Comment (lantoli) [reply examples/mongodbatlas_cluster/tenant-upgrade/README.md]: thanks, changed

File: docs/resources/cluster.md (hunk 5 above)
Comment (lantoli) [reply docs/resources/cluster.md]: thanks, updated

**Review by EspenAlbert — APPROVED** [review]
LGTM

**Review by carriecwk — APPROVED** [review]
Looks good, thank you for making the changes. LGTM!

**Pull request merged**